src/mito2/src/cache/write_cache.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A write-through cache for remote object stores.
+
+use std::sync::Arc;
+
+use object_store::manager::ObjectStoreManagerRef;
+use object_store::ObjectStore;
+use store_api::metadata::RegionMetadataRef;
+use store_api::storage::{RegionId, SequenceNumber};
+use tokio::sync::mpsc::Sender;
+
+use crate::access_layer::sst_file_path;
+use crate::error::Result;
+use crate::read::Source;
+use crate::request::WorkerRequest;
+use crate::sst::file::{FileId, FileMeta, Level};
+use crate::sst::parquet::writer::ParquetWriter;
+use crate::sst::parquet::WriteOptions;
+use crate::wal::EntryId;
+
+/// A cache for uploading files to remote object stores.
+///
+/// It keeps files in local disk and then sends files to object stores.
+pub(crate) struct WriteCache {
```

Comment:
I feel curious about how `WriteCache` and `UploadPartWriter` will interact in the future

Reply:
You can refer to the POC branch.

https://github.com/evenyag/greptimedb/blob/14a325f2275c792d0f495b21b1ca1e1ef37f844d/src/mito2/src/flush.rs#L359-L366